sway-core/src/semantic_analysis/ast_node/declaration/enum.rs

```diff
@@ -27,7 +27,7 @@ impl ty::TyEnumDecl {
         // Type check the type parameters. This will also insert them into the
         // current namespace.
         let new_type_parameters = check!(
-            TypeParameter::type_check_type_params(ctx.by_ref(), type_parameters, true),
+            TypeParameter::type_check_type_params(ctx.by_ref(), type_parameters, false),
```

Comment:
With this, are there any invocations that still require `disallow_trait_constraints` to be true?

Reply:
I added a new issue for this as #4560 as part of #970
Traits are out of scope for this PR but not for implemented trait constraints, please update this PR to reflect that it doesn't close #970

Reply:
I removed #970 from the list of fixed issues by this PR.